### PR TITLE
feat(ui5-barcode-scanner-dialog): add 'open' property

### PR DESCRIPTION
**Pull request by kgogov**
This change introduces a new `open` property to the `ui5-barcode-scanner-dialog` component, enhancing the control over the dialog's visibility. This feature aligns with the behavior of other similar components, providing a more consistent user experience.

Related: #8072

### CODE REVIEW

File: packages/fiori/src/BarcodeScannerDialog.ts [hunk 1]
```diff
@@ -120,6 +120,31 @@ type BarcodeScannerDialogScanErrorEventDetail = {
...
+	}
+
+	set open(value: boolean) {
+		const oldValue = this._open;
+
+		if (oldValue !== value) {
+			this._open = value;
+
+			if (value) {
+				this.show();
```
Comment (ilhan007) [inline packages/fiori/src/BarcodeScannerDialog.ts:141]: Could you try to deprecate show and close in favour of `open` property, and make `open` a regular metadata property.

See `ColorPalletePopover` for reference - 
previously it used to have a `showAt` method (it's kept for compatibility), then we introduced `open` and `opener` and deprecated `showAt`.

Comment (kgogov) [reply packages/fiori/src/BarcodeScannerDialog.ts:141]: Thank you for your comment. I knew that the way I've done it in this pull request is not one of the best practices, so I appreciate your guidance on this.

File: packages/fiori/src/BarcodeScannerDialog.ts [hunk 2]
```diff
@@ -124,21 +134,35 @@ class BarcodeScannerDialog extends UI5Element {
...
 
 	static async onDefine() {
 		BarcodeScannerDialog.i18nBundle = await getI18nBundle("@ui5/webcomponents-fiori");
 	}
 
+	onBeforeRendering() {
+		if (this._previousOpenState !== this.open) {
+			if (this.open) {
+				this.show();
+			} else {
```
Comment (plamenivanov91) [inline packages/fiori/src/BarcodeScannerDialog.ts:155]: Delete this block and add this instead:

```
	get _open() {
		return this.open || undefined;
	}

	onInvalidation(changeInfo: ChangeInfo) {
		if (changeInfo.type === "property" && changeInfo.name === "open") {
			changeInfo.newValue ? this.show() : this.close();
		}
	}
```

**Review by plamenivanov91 — CHANGES_REQUESTED** [review]
In the .hbs file add this to the dialog

`	?open="{{_open}}"
`

Try to add some tests with the newly added code (might simply try the method and the new prop if they get too complex for real-life flow).

**Comment by plamenivanov91** [comment]
> In the .hbs file add this to the dialog
> 
> ` ?open="{{_open}}" `
> 
> Try to add some tests with the newly added code (might simply try the method and the new prop if they get too complex for real-life flow).

By the way with the code I suggested, the `this.dialog.close();` and `this.dialog.show();` become redundant.

File: packages/fiori/src/BarcodeScannerDialog.ts (hunk 2 above)
Comment (kgogov) [reply packages/fiori/src/BarcodeScannerDialog.ts:155]: I really like your suggestion to use `onInvalidation()`. Nice catch!

File: packages/fiori/src/BarcodeScannerDialog.ts [hunk 3]
```diff
@@ -115,6 +115,16 @@ type BarcodeScannerDialogScanErrorEventDetail = {
 })
 
 class BarcodeScannerDialog extends UI5Element {
+	/**
+	 * Indicates whether the dialog is open.
+	 *
+	 * @public
+	 * @default false
+	 * @since 1.23.0
```
[the remark this answers is not in the document]
Comment (kgogov) [reply packages/fiori/src/BarcodeScannerDialog.ts:123]: Fixed.

**Comment by kgogov** [comment]
> > In the .hbs file add this to the dialog
> > `?open="{{_open}}"`
> > Try to add some tests with the newly added code (might simply try the method and the new prop if they get too complex for real-life flow).
> 
> By the way with the code I suggested, the `this.dialog.close();` and `this.dialog.show();` become redundant (in BarcodeScannerDialog.ts).

I considered your suggestion to add `open` attribute in the template but I've encountered the following:
- According to the sequence of actions, we need to first request permission from the user to use the camera and then decide whether to open a dialog. Actually, in our case, whether we put an `open` attribute in the template file is irrelevant, since we use internal methods for the purpose. I tested both cases, noticing that in the first case when we don't put an `open` attribute, in `_showDialog()` we assign a dialog to `this.dialog` with `open` property set to `false` which is correct, but in the second case we receive a dialog with `open` property already set to `true`, which may be not correct, because technically we have not called `this.dialog.show();`
- I'm not so sure that we can remove the logic to show and close the internal managed dialog, because if we do we risk missing the check for user rights to use the camera.

**Review by plamenivanov91 — APPROVED** [review]

**Pull request merged**